app/controllers/devise/saml_sessions_controller.rb

```diff
@@ -17,6 +17,17 @@ def metadata
     render :xml => meta.generate(@saml_config)
   end
 
+  def idp_sign_out
+    if params[:SAMLRequest] && Devise.saml_session_index_key
+      logout_request = OneLogin::RubySaml::SloLogoutrequest.new(params[:SAMLRequest], @saml_config)
+      resource_class.reset_session_key_for(logout_request.name_id)
```

Comment:
Actually, based on the schema, the attribute map is not sent via a logout request.  We only have access to name_id or session index to uniquely identify the user.  We could instead identify the correct user via the session index though.